vet-service/src/test/java/com/petclinic/vet/presentationlayer/VetControllerIntegrationTest.java

```diff
@@ -155,6 +155,45 @@ void deleteARatingForVet_WithValidId_ShouldSucceed() {
                 .expectStatus().isOk()
                 .expectBody();
     }
+
+
+
+    @Test
+    void getAverageRatingByVetId_ShouldSucceed(){
+
+        RatingRequestDTO ratingRequestDTO= RatingRequestDTO.builder()
+                .vetId(vet.getVetId())
+                .rateScore(rating1.getRateScore()).build();
+
+
+        client.get()
+                .uri("/vets/" + ratingRequestDTO.getVetId() + "/ratings" + "/average")
+                .accept(MediaType.APPLICATION_JSON)
+                .exchange()
+                .expectStatus().isOk()
+                .expectHeader().contentType(MediaType.APPLICATION_JSON)
+                .expectBody(Double.class)
+                .value(resp ->{
+                    assertEquals(rating1.getRateScore(), ratingRequestDTO.getRateScore());
+                });
+    }
+
+    @Test
+    void getAverageRatingByVetId_withInvalidVetId_ShouldThrowNotFound(){
+
+        client
+                .get()
+                .uri("/vets/" + INVALID_VET_ID + "/ratings/average")
+                .accept(MediaType.APPLICATION_JSON)
+                .exchange()
+                .expectStatus().isOk()
+                .expectBody(Double.class)
+                .value(avg ->{
+                            assertEquals(0.0,avg);
+                        }
+                );
```

Comment:
Controller Integration test is reactive, looks right. Just the test with invalid vetId could return a NotFound exception.